### DISCUSSION TITLE
API examples for IDEA version 24 + FEA info

### DISCUSSION THREAD

**Discussion by deparser** [idea-statica/ideastatica-public#570] [Help Q&A] [answered]
Hi everyone,

I would like to know if there are any complete examples of the ideaConn API (.ideaConn file with related scripts in Python or C#), as the ones I've seen on [the Idea StatiCa developer site](https://developer.ideastatica.com/) and [the Idea StatiCa GitHub](https://idea-statica.github.io/) seem to be related to versions 21 or lower.

Specifically, my colleagues and I are interested in better understanding the settings of the FEA. I would like to use the functions of the class [IdeaRS.OpenModel.Result](https://idea-statica.github.io/iom/iom-api/latest/html/N_IdeaRS_OpenModel.htm) to analyze the types of finite elements and extract point results. However, I'm unsure if it is possible to investigate this kind of information or if there are limitations.

Thank you in advance,

Dylan

**Comment by mpospisil** [comment]
Hi Dylan, 
We are rewriting API for connections. The new API is based on REST. Python client will be also provided. I believe it we could provide you a beta version in 2 weeks.
Regards, 
Martin

**Comment by nathanlukeIDEA** [comment]
Hi @DevDyl96,

As @mpospisil mentioned we are in the process of updating the connection API and also providing some improved documentation. However the update will work very similar to the current API, with limited increases in functionallity.

However, all the current examples avaliable will work for versions greater than 21. Below is a .zip which provides some basic outline on getting started. 
https://developer.ideastatica.com/examples/readme.html
[C# Training Dataset.zip](https://github.com/user-attachments/files/16121845/C.Training.Dataset.zip)

You are not able to currently extract mesh results through the API. You are only able to get brief results of the calculated connection. Is there a particular reason why you are wanting to get out all the FEA results? 

Thanks 
Nathan

**Reply by deparser** [reply]
Hi @nathanlukeIDEA,
Thank you for your response.

Regarding the FEA results, we aim to transition our internal workflow from Excel to IDEA StatiCa . To facilitate this, we need a comprehensive understanding of the software's behavior and calculation assumptions. For this reason, it will useful for us to compare the IDEA StatiCa results with other software that are well known and used since years. Our goal is not to validate the software but to understand how IDEA StatiCa models a joint compared to our current FEM approach.

Thanks

Dylan

**Reply by mpospisil** [reply, marked as answer]
Hi @DevDyl96 
I understand you, You (as any resposible designer) want to be sure about calculation methods which you use in your designs. On the other hand I would like to say CBFEM is not a new method. It has been using used for 10 years and it has been validated by our experts and also by universities - the validation of the method  is our responsibility. If you have any questions or doubts you can discuss it with our experts. A lot of tests in laboratories were made. I am sure your results can not be more precise then real results from laboratories.
We are open for discussions. The new API will provide you more data (and we believe more user friendly) but a mesh data and FEM results will not be available by API now. 

Thanks,
Martin

**Reply by deparser** [reply]
Thank you @mpospisil and @nathanlukeIDEA for your response.
I will open another discussion to ask some questions about structural checks.

Dylan

**Reply by nathanlukeIDEA** [reply]
Hi @DevDyl96, If you want clarification on IDEA StatiCa calculation methods and structural checks please reach out to your local reseller and or our support team https://www.ideastatica.com/support-center.  If you wish more things for the API then please let us know from further discussion.  :)
